### ISSUE TITLE
Wrong token error while inserting a new link

### ISSUE THREAD

**Issue by hawk291** [shaarli/Shaarli#1371]
Hi, thanks for the app first of all and your time and efforts, it's a pretty tool. 
I'm using Shaarli on raspberry pi. Was working fine (actually all works except insert new link) until recently the error "Wrong token" showed up in the browser, when inserting a new link. I'm using apache and nginx as proxy. 
shaarli_error.log shows this 
`[Wed Oct 09 22:21:28.597912 2019] [php7:notice] [pid 2026] [client 192.168.178.5:41122] PHP Notice:  Undefined index: linkdate in /var/www/shaarli/tmp/linklist.31142c3d4e84edcad5c5c4933542efec.rtpl.php on line 93, referer: https://test.example.com/
Removed all files from /var/www/shaarli/tmp/ to see what happens, still the same error wrong token. 
`
My environment (where shaarli runs)
Raspberry Pi 4, Raspbian GNU/Linux 10 (buster)
Shaarli 0.9.7
PHP version
`PHP 7.3.9-1~deb10u1 (cli) (built: Sep 18 2019 10:33:23) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.9, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.9-1~deb10u1, Copyright (c) 1999-2018, by Zend Technologies
`
Apache version
`Apache/2.4.38 (Raspbian)
`
`nginx version: nginx/1.14.2`

**Comment by ArthurHoaro** [comment]
> Shaarli 0.9.7

You are using an outdated version of Shaarli. Please upgrade to `0.10.x` (stable) or `0.11.x` (latest).

**Comment by ArthurHoaro** [comment]
Closing due to lack of feedback. Feel free to reopon this issue if you still encounter this issue.

**Comment by eazrael** [comment]
Having the same issue. Running 0.12.0-beta after upgrading yesterday PHP  from 7.2.x to 7.4.10.
```
( ! )16 Call Stack #TimeMemoryFunctionLocation 10.0000371408{main}( ).../index.php:0 
20.0055551736Slim\App->run( ).../index.php:151 
30.0068694936Slim\App->process( ).../App.php:297 
40.0068694936Slim\App->callMiddlewareStack( ).../App.php:392 
50.0068694936Slim\App->__invoke( ).../MiddlewareAwareTrait.php:117 
60.0090719272Slim\Route->run( ).../App.php:503 
70.0090720016Slim\Route->callMiddlewareStack( ).../Route.php:268 
80.0090720016Slim\Route->Slim\{closure:/www/ssl/shaarli/vendor/slim/slim/Slim/MiddlewareAwareTrait.php:63-78}( ).../MiddlewareAwareTrait.php:117 
90.0090720016call_user_func:{/www/ssl/shaarli/vendor/slim/slim/Slim/MiddlewareAwareTrait.php:70} Notice: Undefined index: linkdate in /www/ssl/shaarli/tmp/editlink.90100d2eaf5d3705e14b9b4f78ecddc9.rtpl.php on line ( ).../MiddlewareAwareTrait.php:70 
100.0090720016Slim\DeferredCallable->__invoke( ).../MiddlewareAwareTrait.php:70 
110.0091721480call_user_func_array:{/www/ssl/shaarli/vendor/slim/slim/Slim/DeferredCallable.php:57} ( ).../DeferredCallable.php:57 
120.0091721528Shaarli\Front\ShaarliAdminMiddleware->__invoke( ).../DeferredCallable.php:57
130.0091721904Shaarli\Front\ShaarliAdminMiddleware->__invoke( ).../ShaarliAdminMiddleware.php:25 
140.02704909496Slim\Route->__invoke( ).../ShaarliMiddleware.php:54 150.02724912936Slim\Handlers\Strategies\RequestResponse->__invoke( ).../Route.php:281 
160.02724912936 call_user_func:{/www/ssl/shaarli/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php:40} ( ).../RequestResponse.php:40 
170.02724912936Shaarli\Front\Controller\Admin\ManageShaareController->displayCreateForm( ).../RequestResponse.php:40 
180.02744914512Shaarli\Front\Controller\Admin\ManageShaareController->displayForm( ).../ManageShaareController.php:93 
190.04414914528Shaarli\Front\Controller\Admin\ManageShaareController->render( ).../ManageShaareController.php:369 
200.05104915784Shaarli\Render\PageBuilder->render( ).../ShaarliVisitorController.php:66 210.05114915832RainTPL->draw( ).../PageBuilder.php:237 
220.05154936496 include( '/www/ssl/shaarli/tmp/editlink.90100d2eaf5d3705e14b9b4f78ecddc9.rtpl.php' ).../rain.tpl.class.php:190 "> URL

Title

Description

Tags

Type 2 or more characters for results.
```

**Comment by ArthurHoaro** [comment]
Hi @eazrael,

It looks likes there is a line missing in your stacktrace containing the actual error message.
Also a few questions:
  - does this issue happen everytime you try to add a new bookmark?
  - are you also running Shaarli on a raspberry?

**Comment by eazrael** [comment]
I am running a x86_64 Linux. Mostly self-compiled. It's happening all the time, using a chrome-based browser (vivaldi) and the android app shaarlier. 
The stacktrace was in the html form, I had to scrape it from the HTML source code. The missing lines: 

```
Notice: Undefined index: linkdate in /www/ssl/shaarli/tmp/editlink.90100d2eaf5d3705e14b9b4f78ecddc9.rtpl.php on line 16 Call Stack #TimeMemoryFunctionLocation 
10.0000371408{main}( ).../index.php:0 
```

![image](https://user-images.githubusercontent.com/7374919/94276287-52cb3a00-ff48-11ea-839d-d424976aa237.png)

I searched for linkdate and landed in this thread. 

Maybe important is that I upgraded my installation just by overwriting the directory with the github tar ball: 
tar -tvf ../shaarli-v0.12.0-beta-full.tar.gz --strip-components=1
